apiserver/facades/agent/uniter/networkinfo.go

```diff
@@ -183,6 +183,82 @@ func (n *NetworkInfoBase) getEgressForRelation(
 	return subnetsForAddresses(ingress.Values()), nil
 }
 
+func (n *NetworkInfoBase) resolveResultHostNames(netInfoResult params.NetworkInfoResult) params.NetworkInfoResult {
+	// Maintain a cache of host-name -> address resolutions.
+	resolved := make(map[string]string)
+	addressForHost := func(hostName string) string {
+		resolvedAddr, ok := resolved[hostName]
```

Comment:
```suggestion
		resolvedAddr, found := resolved[hostName]
```

Minor nit. Feel free to ignore